src/java/org/apache/cassandra/transport/PreV5Handlers.java

```diff
@@ -63,19 +64,21 @@
          * Note: should only be accessed while on the netty event loop.
          */
         private long channelPayloadBytesInFlight;
-        private boolean paused;
+        
+        /** The cause of the current connection pause, or {@link Overload#NONE} if it is unpaused. */
+        private Overload pauseTrigger = Overload.NONE;
```

Comment:
We're kind of overloading a couple things conceptually here. Consider an `isPaused()` method that checks this variable and renaming this variable to `overloadState` or something. Or not. The change from `paused` to `pauseTrigger` gave me... pause. 😁

Reply:
In early versions of the patch, `paused` was simply removed because it was functionally equivalent to checking `ChannelConfig#isAutoRead()`. So we could add an `isPaused()` method, but I'm not sure what would use it, given `releaseItem()` could not. (i.e. It needs to know why.)

Let me know if we should improve the comment there:

> // The only exception to this is if the global request rate limit has been breached, which means
   // we'll have to wait until a scheduled wakeup task unpauses the connection.

Reply:
Ah. So usage wise, a `pausedDueToByteOverload` or something would be the "only used in one place" context appropriate method, but at that point we're well past diminishing returns into just cruft territory. Fine leaving this as-is. 👍